http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -598,6 +599,53 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
                                            Some(traceId))
         publisherQueue.enqueue1(message).as(message)
       }
+      // we only need to return the message that might cause a start/stop transition
+    } yield msg
+
+  private[service] def processUpdateGceWithPdConfigRequest(req: UpdateRuntimeConfigRequest.GceConfig,
+                                                           allowStop: Boolean,
+                                                           runtime: Runtime,
+                                                           gceConfig: RuntimeConfig.GceWithPdConfig,
+                                                           traceId: TraceId): F[Option[UpdateRuntimeMessage]] =
+    for {
+      // should machine type be updated?
+      targetMachineType <- getTargetMachineType(gceConfig.machineType,
+                                                req.updatedMachineType,
+                                                runtime.projectNameString,
+                                                runtime.status,
+                                                allowStop)
+      // should disk size be updated?
+      diskId <- F.fromEither(
+        gceConfig.persistentDiskId.toRight(RuntimeDiskNotFound(runtime.googleProject, runtime.runtimeName, traceId))
+      )
+      diskOpt <- persistentDiskQuery.getById(diskId).transaction
+      disk <- F.fromEither(diskOpt.toRight(DiskNotFoundByIdException(diskId, traceId)))
+      targetDiskSize <- traverseIfChanged(req.updatedDiskSize, disk.size) { d =>
+        if (d.gb < disk.size.gb)
+          Async[F].raiseError[DiskSize](RuntimeDiskSizeCannotBeDecreasedException(runtime.projectNameString))
+        else if (!allowStop)
+          Async[F].raiseError[DiskSize](RuntimeDiskSizeCannotBeChangedException(runtime.projectNameString))
+        else
+          Async[F].pure(d)
+      }
+
+      // if target disk size is defined, enqueue pubsub messages to change the PD size and restart the runtime
+      _ <- targetDiskSize.traverse { size =>
+        val diskMessage = UpdateDiskMessage(diskId, size, Some(runtime.id), Some(traceId))
```

Comment:
why don't we merge this intent in `UpdateRuntimeMessage` instead of a separate message for updating diskMesage? won't be get in race condition this way since `UpdateRuntimeMessage` can potentially trigger a restart as well?